src/vm/types/mod.rs

```diff
@@ -330,31 +554,126 @@ impl Value {
         //     this is a problem _if_ the static analyzer cannot already prevent
         //     this case. This applies to all the constructor size checks.
         let type_sig = TypeSignature::construct_parent_list_type(&list_data)?;
-        Ok(Value::List(ListData { data: list_data, type_signature: type_sig }))
+        Ok(Value::Sequence(SequenceData::List(ListData { data: list_data, type_signature: type_sig })))
     }
 
     pub fn buff_from(buff_data: Vec<u8>) -> Result<Value> {
         // check the buffer size
         BufferLength::try_from(buff_data.len())?;
         // construct the buffer
-        Ok(Value::Buffer(BuffData { data: buff_data }))
+        Ok(Value::Sequence(SequenceData::Buffer(BuffData { data: buff_data })))
     }
 
     pub fn buff_from_byte(byte: u8) -> Value {
-        Value::Buffer(BuffData { data: vec![byte] })
+        Value::Sequence(SequenceData::Buffer(BuffData { data: vec![byte] }))
+    }
+
+    pub fn string_ascii_from_bytes(bytes: Vec<u8>) -> Result<Value> {
+        for b in bytes.iter() {
+            if !b.is_ascii_alphanumeric() && !b.is_ascii_punctuation() && !b.is_ascii_whitespace() {
+                return Err(CheckErrors::InvalidCharactersDetected.into());
+            }
+        }
+        // check the string size
+        BufferLength::try_from(bytes.len())?;
+        // construct the string        
+        Ok(Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data: bytes }))))
+    }
+
+    pub fn string_utf8_from_string_utf8_literal(tokenized_str: String) -> Result<Value> {
+        let wrapped_codepoints_matcher = Regex::new("^\\\\u\\{(?P<value>[[:xdigit:]]+)\\}").unwrap();
+        let mut window = tokenized_str.as_str();
+        let mut cursor = 0;
+        let mut data: Vec<Vec<u8>> = vec![];
+        while !window.is_empty() {
```

Comment:
You should probably also add a condition here to make sure that `cursor` does not exceed the maximum string length.  Otherwise, this code can keep looping and adding bytes to `data` well beyond 4 GB.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1779/commits/6677278149912738bfe8f7e74c758a7fc4c24d97